### ISSUE TITLE
Generic Marker Widget

### ISSUE THREAD

**Issue by svenefftinge** [theia-ide/theia#221]
A generic widget shall be provided for extension authors to build widgets that display markers of a certain kind. A context menu shall allow to register actions on selected items.
This list view shall form the foundation for the Problems View (#26)  and the Bookmarks View(#222).

**Comment by svenefftinge** [comment]
It was done with with https://github.com/theia-ide/theia/pull/455

**Comment by hexa00** [comment]
In the original post there there is mention of a context menu to register actions is that present?

**Comment by svenefftinge** [comment]
It needs to be configured per concrete marker widget. E.g. for the ProblemsView a context menu with the prefix 'marker-context-menu' is registered. I will reword it to 'problems' (incl. the variables).
